rasa/core/featurizers.py

```diff
@@ -258,15 +261,15 @@ def create_encoded_all_actions(self, domain: Domain) -> np.ndarray:
 
 
 class TrackerFeaturizer(object):
-    """Base class for actual tracker featurizers"""
+    """Base class for actual tracker featurizers."""
 
     def __init__(
         self,
         state_featurizer: Optional[SingleStateFeaturizer] = None,
         use_intent_probabilities: bool = False,
     ) -> None:
 
-        self.state_featurizer = state_featurizer or SingleStateFeaturizer()
+        self.state_featurizer = state_featurizer
```

Comment:
`MemoizationPolicy` doesn't use state featurizer therefore, I think it is better to set it to `None` instead of a base class. I added error if it is called and `None`. Is `TypeError` a correct error?

Reply:
Please use `ValueError`

Reply:
👍